test/new-e2e/tests/npm/ec2_1host_test.go

```diff
@@ -92,6 +98,66 @@ func (v *ec2VMSuite) TestFakeIntakeNPM() {
 	}, 90*time.Second, time.Second, "not enough connections received")
 }
 
+// TestFakeIntakeNPM_600pluscnx Validate the agent can communicate with the (fake) backend and send connections
+// every 30 seconds with a maximum of 600 connections per payloads, if more another payload will follow.
+//   - looking for 1 host to send CollectorConnections payload to the fakeintake
+//   - looking for n payloads and check if the last 2 have a maximum span of 100ms
+func (v *ec2VMSuite) TestFakeIntakeNPM_600pluscnx() {
+	t := v.T()
+
+	targetHostnameNetID := ""
+	// looking for 1 host to send CollectorConnections payload to the fakeintake
+	v.EventuallyWithT(func(c *assert.CollectT) {
+		// generate a connection
+		v.Env().RemoteHost.MustExecute("ab -n 600 -c 600 http://www.datadoghq.com/")
+
+		hostnameNetID, err := v.Env().FakeIntake.Client().GetConnectionsNames()
+		assert.NoError(c, err, "GetConnectionsNames() errors")
+		if !assert.NotEmpty(c, hostnameNetID, "no connections yet") {
+			return
+		}
+		targetHostnameNetID = hostnameNetID[0]
+
+		t.Logf("hostname+networkID %v seen connections", hostnameNetID)
+	}, 60*time.Second, time.Second, "no connections received")
+
+	// looking for x payloads (with max 600 connections) and check if the last 2 have a max span of 100ms
+	v.EventuallyWithT(func(c *assert.CollectT) {
+		cnx, err := v.Env().FakeIntake.Client().GetConnections()
+		assert.NoError(t, err)
+
+		if !assert.Greater(c, len(cnx.GetPayloadsByName(targetHostnameNetID)), 2, "not enough payloads") {
+			return
+		}
+
+		cnx.ForeachConnection(func(c *agentmodel.Connection, cc *agentmodel.CollectorConnections, hostname string) {
+			assert.LessOrEqualf(t, len(cc.Connections), 600, "too many payloads")
+		})
+
+		found600plusConnections := false
+		hostPayloads := cnx.GetPayloadsByName(targetHostnameNetID)
+		lenHostPayloads := len(hostPayloads)
+		for i, cc := range hostPayloads {
```

Comment:
Do we need this loop? Wouldn't checking `hostPayloads[lenHostPayloads-2].Connections` be enough?

Reply:
yes it's enough, I tried to do 2 things in the loop in previous iteration
thanks